### PR TITLE
[BDRSPS-896] Added `template_status` to template metadata schema

### PR DESCRIPTION
**Pull request by chungvl**

### CODE REVIEW

File: abis_mapping/types/metadata.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ class TemplateMetadata(pydantic.BaseModel):
     sampling_type: str
     template_url: str
     schema_url: str
+    template_status: str
```
Comment (serge-gaia) [inline abis_mapping/types/metadata.py:26]: I'm trying to find a better name, like 'lifecycle_status` or `support_status`.
Also I think the type should be an `Enum`, with the values:
* "Beta"
* "Current"
* "Deprecated"

Comment (serge-gaia) [reply abis_mapping/types/metadata.py:26]: I opt for `lifecycle_status`

Comment (chungvl) [reply abis_mapping/types/metadata.py:26]: The phrase `lifecycle` is going to get confusing given we're already talking about biology?

File: abis_mapping/types/metadata.py [hunk 2]
```diff
@@ -23,6 +24,7 @@ class TemplateMetadata(pydantic.BaseModel):
     sampling_type: str
     template_url: str
     schema_url: str
+    template_lifecycle_status: Literal["Beta", "Current", "Deprecated"]
```
Comment (serge-gaia) [inline abis_mapping/types/metadata.py:27]: An `Enum` would have been good here, I think.

Comment (serge-gaia) [reply abis_mapping/types/metadata.py:27]: But that's fine, merging.

**Review by serge-gaia — APPROVED** [review]

**Pull request merged**